### PR TITLE
fix: simplify README, remove AWS Glue for Ray references

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Docs

### Detail
- Refresh README
- Remove references to AWS Glue for Ray

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 516a5a2a0be2e8e24e0fd84ea4229df49704acec
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d8d25bd3-f997-4ee3-a6c1-d1021061dc51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 516a5a2a0be2e8e24e0fd84ea4229df49704acec
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a8d921b4-6bd9-44af-8b99-81cf2e7d41f7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: README.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
 # AWS SDK for pandas (awswrangler)
 
-AWS Data Wrangler is now **AWS SDK for pandas (awswrangler)**.  We’re changing the name we use when we talk about the library, but everything else will stay the same.  You’ll still be able to install using `pip install awswrangler` and you won’t need to change any of your code.  As part of this change, we’ve moved the library from AWS Labs to the main AWS GitHub organisation but, thanks to the GitHub’s redirect feature, you’ll still be able to access the project by its old URLs until you update your bookmarks.  Our documentation has also moved to [aws-sdk-pandas.readthedocs.io](https://aws-sdk-pandas.readthedocs.io), but old bookmarks will redirect to the new site.
-
 *Pandas on AWS*
 
 Easy integration with Athena, Glue, Redshift, Timestream, OpenSearch, Neptune, QuickSight, Chime, CloudWatchLogs, DynamoDB, EMR, SecretManager, PostgreSQL, MySQL, SQLServer and S3 (Parquet, CSV, JSON and EXCEL).
 
-![AWS SDK for pandas](docs/source/_static/logo2.png?raw=true "AWS SDK for pandas")
+![AWS SDK for pandas](https://github.com/aws/aws-sdk-pandas/blob/main/docs/source/_static/logo2.png?raw=true "AWS SDK for pandas")
```
Comment (LeonLuttenberger) [inline README.md:7]: Don't we want this to be a "local" path. For example, if I'm on the `release/2.x.` branch, I would want the readme to display that version of the logo.

File: docs/source/scale.rst [hunk 2]
```diff
@@ -3,8 +3,6 @@ At scale
 
 AWS SDK for pandas supports `Ray <https://www.ray.io/>`_ and `Modin <https://modin.readthedocs.io/en/stable/>`_, enabling you to scale your pandas workflows from a single machine to a multi-node environment, with no code changes.
 
-The simplest way to try this is with `AWS Glue for Ray <https://aws.amazon.com/blogs/big-data/introducing-aws-glue-for-ray-scaling-your-data-integration-workloads-using-python/>`_, the new serverless option to run distributed Python code announced at AWS re:Invent 2022. AWS SDK for pandas also supports self-managed Ray on `Amazon Elastic Compute Cloud (Amazon EC2) <https://github.com/aws/aws-sdk-pandas/blob/main/tutorials/035%20-%20Distributing%20Calls%20on%20Ray%20Remote%20Cluster.ipynb>`_.
```
Comment (LeonLuttenberger) [inline docs/source/scale.rst:6]: We could keep the section about how to use self-managed Ray on EC2.

File: README.md [hunk 3]
```diff
@@ -228,36 +208,3 @@ Into AWS lambda:
 import logging
 logging.getLogger("awswrangler").setLevel(logging.DEBUG)
 ```
-
-## Who uses AWS SDK for pandas?
```
Comment (LeonLuttenberger) [inline README.md:232]: Are we getting rid of this entirely? If so, we should get rid of the [issue template](https://github.com/aws/aws-sdk-pandas/blob/main/.github/ISSUE_TEMPLATE/support-data-wrangler.md) as well.

File: README.md (hunk 1 above)
Comment (jaidisido) [reply README.md:7]: I was just being consistent with the rest of the [README](https://github.com/aws/aws-sdk-pandas/blob/main/README.md?plain=1#L125) but perhaps we should make the change everywhere else

File: README.md (hunk 3 above)
Comment (jaidisido) [reply README.md:232]: This is up for discussion. I think the library is popular enough these days that we don't need public endorsing that can be outdated. But yes good point on the issue template, will delete it

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e66ec562d0340fb8dd0b02d61a70b2f094bbf393
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=108ae656-0490-4b6f-a611-b1ad14ca9ebf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e66ec562d0340fb8dd0b02d61a70b2f094bbf393
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e4cec500-00b0-409a-b90e-27f5e7395a63%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 03bf2773816c4d56b293837811550a4930c56ac0
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=68f44167-31b7-4706-b90c-8eed4dc6dc7f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 8b1c246ddcc5c4cf4e98845befb841e91492c9b3
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f69da28e-0d08-4281-9a8e-8d66f32028b9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**